telegram/ext/jobqueue.py

```diff
@@ -195,6 +196,111 @@ def run_repeating(self, callback, interval, first=None, context=None, name=None)
+        """
+        dt = datetime.datetime.now(tz=when.tzinfo or _UTC)
+        next_dt = dt
+        dt_time = dt.time().replace(tzinfo=when.tzinfo)
+        if calendar.monthrange(dt.year, dt.month)[1] < day:
+            # if the day does not exist in the current month (e.g Feb 31st)
+            if day_is_strict is False:
+                # set day as last day of month instead
+                next_dt = dt + datetime.timedelta(days=calendar.monthrange(
+                                                  dt.year, dt.month)[1] - dt.day)
+            else:
+                # else set as day in subsequent month. Subsequent month is
+                # guaranteed to have the date, if current month does not have the date.
+                next_dt = dt + datetime.timedelta(days=calendar.monthrange(
+                                                  dt.year, dt.month)[1] - dt.day + day)
+        else:
+            # if the day exists in the current month
+            if dt.day < day:
+                # day is upcoming
+                next_dt = dt + datetime.timedelta(day - dt.day)
+            elif dt.day > day or (dt.day == day and dt_time > when):
+                # run next month if day has already passed
+                next_dt = dt + datetime.timedelta(days=calendar.monthrange(
+                                                  dt.year, dt.month)[1] - dt.day + day)
+                # do not need to account for Dec - Jan because they both have 31 days
+                next_month_has_date = next_dt.month - dt.month == 1
+                if next_month_has_date is False and day_is_strict is False:
+                    # schedule in the next month last date if day is not strict
+                    next_dt = dt + datetime.timedelta(days=calendar.monthrange(
+                                                      dt.year, dt.month)[1] - dt.day
+                                                      + calendar.monthrange(dt.year + 1
+                                                                            if dt.month == 12
+                                                                            else dt.year,
+                                                                            1 if dt.month == 12
+                                                                            else dt.month + 1)[1])
+                elif next_month_has_date is False and day_is_strict:
+                    # schedule the subsequent month if day is strict
+                    next_dt = dt + datetime.timedelta(
+                        days=calendar.monthrange(dt.year, dt.month)[1]
+                        - dt.day + calendar.monthrange(
+                            dt.year + 1 if
+                            dt.month == 12
+                            else dt.year,
+                            1 if dt.month == 12
+                            else dt.month + 1)[1] + day)
+
+        next_dt = next_dt.replace(hour=when.hour, minute=when.minute, second=when.second,
```

Comment:
Is there a reason you removed fold?

Reply:
yeah, the argument `fold` is not present in older versions of python

Reply:
Oh okay. Can you test with `hasattribute` and replace if `fold` exists as an attribute?

Reply:
done!